workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/InviteMemberModal/InviteMemberModal.tsx

```diff
@@ -220,6 +290,23 @@ export function InviteMemberModal({
 
 			return role;
 		}, []);
+
+		const rolesSelected: string[] = rolesChecked
+			.filter(
+				(roleCheck) =>
+					roleCheck.isChecked &&
+					(accountRoles || []).some(
+						(accountRole) => roleCheck.roleName === accountRole.name
+					)
+			)
+			.map((roleCheck) => roleCheck.roleName);
+
+		if (!rolesChecked.length || !rolesSelected.length) {
+			setError('roles', {message: 'Please select at least one role'});
+		}
+
+		setValue('roles', rolesSelected as any);
```

Comment:
Hey @juliocezar96, I think you can create a function just to check if the roles are selected, filter these roles and list.
I was thinking something like this:

```
function isRoleSelected(roleCheck, accountRoles) {
return roleCheck.isChecked && (accountRoles?.some((accountRole) => roleCheck.roleName === accountRole.name) ?? false)};
```

Then you can filter and get the list like you did:
`const rolesSelected = rolesChecked.filter((roleCheck) => isRoleSelected(roleCheck, accountRoles)).map((roleCheck) => roleCheck.roleName)`

This way we can break the function to be more descriptive and readble. What do you think? Makes sense?